cmd/lava/internal/scan/scan.go

```diff
@@ -140,8 +140,8 @@ func scan(args []string) (int, error) {
 	metrics.Collect("exit_code", exitCode)
 	metrics.Collect("duration", time.Since(startTime).Seconds())
 
-	if cfg.ReportConfig.Metrics != "" {
-		if err = metrics.WriteFile(cfg.ReportConfig.Metrics); err != nil {
+	if config.Get(cfg.ReportConfig.Metrics) != "" {
+		if err = metrics.WriteFile(config.Get(cfg.ReportConfig.Metrics)); err != nil {
```

Comment:
You can avoid one call to `config.Get` with:

```suggestion
	if metricsFile := config.Get(cfg.ReportConfig.Metrics); metricsFile != "" {
		if err = metrics.WriteFile(metricsFile); err != nil {
```